php-frontend/src/main/java/org/sonar/php/tree/impl/SeparatedListImpl.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * SonarQube PHP Plugin
+ * Copyright (C) 2010 SonarSource and Akram Ben Aissi
+ * sonarqube@googlegroups.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonar.php.tree.impl;
+
+import com.google.common.base.Function;
+import com.google.common.base.Functions;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.UnmodifiableIterator;
+import org.sonar.php.tree.impl.lexical.InternalSyntaxToken;
+import org.sonar.plugins.php.api.tree.SeparatedList;
+import org.sonar.plugins.php.api.tree.Tree;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+
+public class SeparatedListImpl<T extends Tree> implements SeparatedList<T> {
+
+  private final List<T> list;
+  private final List<InternalSyntaxToken> separators;
+
+  public SeparatedListImpl(List<T> list, List<InternalSyntaxToken> separators) {
+    Preconditions.checkArgument(
+      list.size() == separators.size() + 1 || list.size() == separators.size(),
+      "Instanciating a SeparatedList with inconsistent number of elements (%s) and separators (%s)",
+      list.size(), separators.size());
+    this.list = list;
+    this.separators = separators;
+  }
+
+  public static <T extends Tree> SeparatedListImpl<T> empty() {
+    return new SeparatedListImpl<>(ImmutableList.<T>of(), ImmutableList.<InternalSyntaxToken>of());
+  }
+
+  public InternalSyntaxToken getSeparator(int i) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the "@Override" annotation above this method signature [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1161)